src/main/java/com/azure/cosmos/examples/modeling/async/ModelingDemos.java

```diff
@@ -0,0 +1,637 @@
+                    exit = true;
+                }
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error(String.format("Cosmos getStarted failed with %s", e));
+        } finally {
+            logger.info("Closing the client");
+            p.shutdown();
+        }
+    }
+
+    ModelingDemos() {
+        ArrayList<String> preferredRegions = new ArrayList<>();
+        preferredRegions.add("West US");
+        client = new CosmosClientBuilder()
+                .endpoint(AccountSettings.HOST)
+                .key(AccountSettings.MASTER_KEY)
+                .preferredRegions(preferredRegions)
+                .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                .contentResponseOnWriteEnabled(true)
+                .buildAsyncClient();
+
+    }
+    // </Main>
+
+    public void QueryCustomer() {
+        database = client.getDatabase("database-v2");
+        container = database.getContainer("customer");
+        int preferredPageSize = 10;
+        CosmosQueryRequestOptions queryOptions = new CosmosQueryRequestOptions();
+        queryOptions.setQueryMetricsEnabled(true);
+
+        String customerId = "FFD0DD37-1F0E-4E2E-8FAC-EAF45B0E9447";
+        CosmosPagedFlux<Product> customerPagedFluxResponse = container.queryItems(
+                "SELECT * FROM c WHERE c.id = \"" + customerId + "\"", queryOptions, Product.class);
+
+        try {
+            customerPagedFluxResponse.byPage(preferredPageSize).flatMap(fluxResponse -> {
+                logger.info("Got a page of query result with " +
+                        fluxResponse.getResults().size() + " items(s)"
+                        + " and request charge of " + fluxResponse.getRequestCharge());
+                logger.info("Item Ids " + fluxResponse
+                        .getResults()
+                        .stream()
+                        .map(Product::getId)
+                        .collect(Collectors.toList()));
+                return Flux.empty();
+            }).blockLast();
+
+        } catch (Exception err) {
+            if (err instanceof CosmosException) {
+                // Client-specific errors
+                CosmosException cosmosErr = (CosmosException) err;
+                cosmosErr.printStackTrace();
+                logger.error(String.format("Read Item failed with %s\n", cosmosErr));
+            } else {
+                // General errors
+                err.printStackTrace();
+            }
+        }
+    }
+
+    public void GetCustomer() {
+        try {
+            database = client.getDatabase("database-v2");
+            container = database.getContainer("customer");
+            String customerId = "FFD0DD37-1F0E-4E2E-8FAC-EAF45B0E9447";
+            Mono<CosmosItemResponse<CustomerV2>> item = container.readItem(customerId, new PartitionKey(customerId),
+                    CustomerV2.class);
+            double requestCharge = item.block().getRequestCharge();
```

Comment:
Calling block twice will trigger the requests twice. so the requestCharge and requestLatency are not for the same request. 

we can change to following:
```
CosmosItemResponse<CustomerV2> item = container.readItem(customerId, new PartitionKey(customerId),
                     CustomerV2.class).block();
             double requestCharge = item.getRequestCharge();
            Duration requestLatency = item.getDuration();

```